server/app/views/admin/programs/ProgramBlockReadOnlyView.java

```diff
@@ -0,0 +1,312 @@
+    DivTag mainContent =
+        div()
+            .withClasses(
+                "flex", "flex-grow", "flex-col", "px-2", StyleUtils.responsive2XLarge("px-16"))
+            .with(
+                renderProgramInfo(programDefinition),
+                div()
+                    .withClasses("flex", "flex-grow", "-mx-2")
+                    .with(blockOrderPanel(request, programDefinition, blockDefinition.id()))
+                    .with(blockPanel(programDefinition, blockDefinition, blockForm, questions)));
+
+    return layout
+        .getBundle()
+        .setTitle(String.format("View %s", blockDefinition.name()))
+        .addMainContent(mainContent);
+  }
+
+  /*
+   * Creates the panel that shows a list of all screens and allows reordering them as well as
+   * selecting one to see the details of the screen definition.
+   */
+  protected DivTag blockOrderPanel(
+      Request request, ProgramDefinition programDefinition, long focusedBlockId) {
+    return div()
+        .withClasses("shadow-lg", "pt-6", "w-2/12", "border-r", "border-gray-200")
+        .with(
+            blockList(
+                request,
+                programDefinition,
+                programDefinition.getNonRepeatedBlockDefinitions(),
+                focusedBlockId,
+                0));
+  }
+
+  /*
+   * Recursively renders a list of all blocks of a given program to be shown in the block
+   * order panel.
+   */
+  private DivTag blockList(
+      Request request,
+      ProgramDefinition programDefinition,
+      ImmutableList<BlockDefinition> blockDefinitions,
+      long focusedBlockId,
+      int level) {
+    DivTag container = div().withClass("pl-" + level * 2);
+    int index = 0;
+    for (BlockDefinition blockDefinition : blockDefinitions) {
+      container.with(blockTag(blockDefinitions, index++, programDefinition, focusedBlockId));
+
+      // Recursively add repeated blocks indented under their enumerator block
+      if (blockDefinition.isEnumerator()) {
+        container.with(
+            blockList(
+                request,
+                programDefinition,
+                programDefinition.getBlockDefinitionsForEnumerator(blockDefinition.id()),
+                focusedBlockId,
+                level + 1));
+      }
+    }
+    return container;
+  }
+
+  /*
+   * Creates a Div that represents an individual block in the  list of blocks shown in the
+   * block order panel.
+   */
+  protected DivTag blockTag(
+      ImmutableList<BlockDefinition> blockDefinitions,
+      int blockIndex,
+      ProgramDefinition programDefinition /* needed for subclasses */,
+      long focusedBlockId) {
+    //
+    BlockDefinition blockDefinition = blockDefinitions.get(blockIndex);
+    int numQuestions = blockDefinition.getQuestionCount();
+    String selectionColoring = blockDefinition.id() == focusedBlockId ? "bg-gray-100" : "";
+    String questionCountText = String.format("Question count: %d", numQuestions);
+
+    String blockLink =
```

Comment:
Yes that is the intent eventually, however, this cl is only a step in the direction and the ReadOnlyView is only used through its child, the edit view. 
The AdminProgramBlocksController creates the screen and calls the editView in that case.

The correct interactive behaviour for a read only version is not yet implemented and will be part of a separate cl.
It's a good hint for why something may not be working for that change though. I've added a todo in the code based on your comment to remind myself.

Reply:
I'm hesitant to check in knowingly broken code.  this url will result in a 400 or a 500 result for instance.

If we really need to check this in as is can you make the constructor throw an UnsupportedOperationException exception with comment, and make this url a clearly incorrect one rather than something that looks like it does work?

String blockLink = "notyetimplementeditlink"

Reply:
Good suggestion. Done.